pkg/sql/crdb_internal.go

```diff
@@ -354,6 +355,72 @@ CREATE TABLE crdb_internal.databases (
 	},
 }
 
+var crdbInternalExternalConnections = virtualSchemaTable{
+	comment: `list external connections that can be used by the current user`,
+	schema: `
+CREATE TABLE crdb_internal.external_connections (
+	name STRING NOT NULL,
+	created TIMESTAMP NOT NULL,
+	updated TIMESTAMP NOT NULL,
+	connection_type STRING NOT NULL,
+	connection_details JSONB NOT NULL
+)`,
+	populate: func(ctx context.Context, p *planner, db catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		// Get all external connections in the cluster.
+		query := `
+SELECT
```

Comment:
Would we also want OWNER here? (maybe not?)

Reply:
it probably should similar to how `SHOW SCHEDULES` does.